Plugins/BuildServerIntegration/AzureDevOpsIntegration/AzureDevOpsAdapter.cs

```diff
@@ -122,15 +139,69 @@ private async Task ObserveBuildsAsync(DateTime? sinceDate, bool running, IObserv
             {
                 if (_buildDefinitions == null)
                 {
-                    _buildDefinitions = await _buildDefinitionsTask.JoinAsync();
+                    try
+                    {
+                        _buildDefinitions = await _buildDefinitionsTask.JoinAsync();
+
+                        if (_buildDefinitions == null)
+                        {
+                            observer.OnCompleted();
+                            return;
+                        }
 
-                    if (_buildDefinitions == null)
+                        CacheAzureDevOps = new CacheAzureDevOps { Id = CacheKey, BuildDefinitions = _buildDefinitions };
+                    }
+                    catch (UnauthorizedAccessException)
+                    {
+                        string errorMessage = _badTokenErrorMessage.Text;
+                        ManageError(errorMessage);
+
+                        if (ProjectOnErrorKey == null || ProjectOnErrorKey != CacheKey)
+                        {
+                            ProjectOnErrorKey = CacheKey;
+
+                            var btnUpdateToken = new TaskDialogButton("btnUpdateToken", "Update token");
+                            btnUpdateToken.Click += BtnUpdateToken_Click;
+                            var btnIgnoreError = new TaskDialogButton("btnIgnoreError", "Ignore error");
```

Comment:
```suggestion
                            var btnIgnoreError = new TaskDialogButton("btnIgnoreError", "Ignore");
```

Reply:
I have renamed the "Update token" in "Open settings" (see the updated screenshot in the first message) because:
1. That's what is done in fact
2. That allows the taskdialog to be reused for other reasons where we detect a bad configuration. For example if no build definitions are found...